src/geom/SpanSet.cc

```diff
@@ -635,32 +717,69 @@ std::shared_ptr<SpanSet> SpanSet::intersectNot(SpanSet const& other) const {
     if (!getBBox().overlaps(other.getBBox())) {
         return std::make_shared<SpanSet>(this->begin(), this->end());
     }
+    // Handle calling a SpanSet's intersectNot with itself as an argument
+    if (other == *this) {
+        return std::make_shared<SpanSet>();
+    }
     /* This function must find all the areas in this and not in other. These SpanSets
      * may be overlapping with this less than other a1|    b1|   a2|    b2|,
      * with this greater than other b1|    a1|     b2|    a2|,
      * with this containing other a1|    b1|     b2|     a2|,
      * or with other containing this  b1|    a1|    a2|   b2|
      */
     std::vector<Span> tempVec;
-    bool added;
+    auto otherIter = other.begin();
     for (auto const& spn : _spanVector) {
-        added = false;
-        for (auto const& otherSpn : other) {
-            if (spansOverlap(spn, otherSpn)) {
+        bool added = false;
+        bool spanStarted = false;
+        int spanBottom = 0;
+        while (otherIter != other.end() && otherIter->getY() <= spn.getY()) {
+            if (spansOverlap(spn, *otherIter)) {
                 added = true;
```

Comment:
So this will go wrong in case `other` fully contains `spn` (nothing is added).
Perhaps it is best in the unit test to generate many cases at random (once) and compare them with a brute force comparison.
It doesn't seem possible to device a full set of test cases.